Pandas_HW.py

```diff
@@ -0,0 +1,174 @@
+
+logFC_sorted = sorted(logFC)
+
+A_min = logFC_sorted[0]
+B_min = logFC_sorted[1]
+print(A_min, B_min)
+
+A_max = logFC_sorted[-1]
+B_max = logFC_sorted[-2]
+print(A_max, B_max)
+
+
+# ### Находим значения p-value, соответствующие полученным значениям logFC:
+
+Ap_min = diffexpr_data[diffexpr_data['logFC'] == A_min]['log_pval']
+A_pval_min = Ap_min.to_list()[0]
+
+Bp_min = diffexpr_data[diffexpr_data['logFC'] == B_min]['log_pval']
+B_pval_min = Bp_min.to_list()[0]
+
+Ap_max = diffexpr_data[diffexpr_data['logFC'] == A_max]['log_pval']
+A_pval_max = Ap_max.to_list()[0]
+
+Bp_max = diffexpr_data[diffexpr_data['logFC'] == B_max]['log_pval']
+B_pval_max = Bp_max.to_list()[0]
+
+
+# ### Находим названия генов:
+
+Ag_min = diffexpr_data[diffexpr_data['logFC'] == A_min]['Sample']
+A_gene_min = Ag_min.to_list()[0]
+
+Bg_min = diffexpr_data[diffexpr_data['logFC'] == B_min]['Sample']
+B_gene_min = Bg_min.to_list()[0]
+
+Ag_max = diffexpr_data[diffexpr_data['logFC'] == A_max]['Sample']
+A_gene_max = Ag_max.to_list()[0]
+
+Bg_max = diffexpr_data[diffexpr_data['logFC'] == B_max]['Sample']
+B_gene_max = Bg_max.to_list()[0]
+
+
+# ### Строим violent plot:
+
+fig, ax = plt.subplots(figsize=(25, 15))
+plt.rc('font')
+
+down_left = diffexpr_data[(diffexpr_data['logFC']<=0)&(diffexpr_data['log_pval']<=0.05)]
+up_left = diffexpr_data[(diffexpr_data['logFC']<=0)&(diffexpr_data['log_pval']>=0.05)]
+down_right = diffexpr_data[(diffexpr_data['logFC']>=0)&(diffexpr_data['log_pval']<=0.05)]
+up_right = diffexpr_data[(diffexpr_data['logFC']>=0)&(diffexpr_data['log_pval']>=0.05)]
+
+plt.scatter(x=down_left['logFC'],y=down_left['log_pval'],s=20,label="Non-significantly downregulated",color="green")
+plt.scatter(x=up_left['logFC'],y=up_left['log_pval'],s=20,label="Significantly downregulated",color="blue")
+plt.scatter(x=down_right['logFC'],y=down_right['log_pval'],s=20,label="Non-significantly upregulated",color="red")
+plt.scatter(x=up_right['logFC'],y=up_right['log_pval'],s=20,label="Significantly upregulated",color="orange")
+
+# Устанавливаем название и параметры осей
+#plt.rc('axes', linewidth=4)
+plt.xlabel(r'$\bf{log_2}$ (fold change)', size = 25, fontweight = 'bold', style='italic')
+plt.ylabel(r'$\bf{- log_{10}}$ (p-value corrected)', size = 25, fontweight = 'bold', style='italic')
+plt.rcParams["axes.labelweight"] = "bold"
+
+# Устанавливаем ticks
+plt.xticks(size = 20, fontweight = 'bold')
+plt.yticks(size = 20, fontweight = 'bold')
+plt.minorticks_on()
+plt.tick_params(axis="x", which="minor", length=5, width=3)
+plt.tick_params(axis="y", which="minor", length=5, width=3)
+plt.tick_params(axis="x", which="major", length=5, width=3)
+plt.tick_params(axis="y", which="major", length=5, width=3)
+
+# Название графика:
+plt.title("Volcano plot", size = 40, fontweight = 'bold', style='italic')
+
+# Пунктирные линии:
+plt.axvline(0,color="grey",linestyle="--", linewidth = 3)
+plt.axhline(0.05,color="grey",linestyle="--", linewidth = 3)
```

Comment:
Ось ординат у тебя в шкале $-log_{10}(p\ value)$, а значит и линия должна соответствовать
```suggestion
plt.axhline(-np.log10(0.05) ,color="grey",linestyle="--", linewidth = 3)
```